code/modules/urist/items/clothstuff.dm

```diff
@@ -94,3 +94,46 @@
 				else
 					to_chat(user, "<span class='warning'>\The [src] is used up, but there are more wounds to treat on \the [affecting.name].</span>")
 			use(used)
+
+/obj/item/loom
+	name = "table loom"
+	desc = "A loom small enough to only take up a table instead of the whole floor."
+	icon = 'icons/urist/items/wood.dmi'
+	icon_state = "loom"
+
+/datum/reagent/cottonfiber
+	name = "cotton fiber"
+	description = "A mass of cotton fibers."
+	taste_description = "cotton"
+	reagent_state = SOLID
+	color = "#ffffff"
+
+/datum/seed/cotton
+	name = "cotton"
+	seed_name = "cotton"
+	display_name = "cotton plant"
+	chems = list(/datum/reagent/cottonfiber = list(6,1))
+
+/datum/seed/cotton/New()
+	..()
+	set_trait(TRAIT_MATURATION,10)
+	set_trait(TRAIT_PRODUCTION,1)
+	set_trait(TRAIT_YIELD,7)
+	set_trait(TRAIT_POTENCY,10)
+	set_trait(TRAIT_PRODUCT_ICON,"cotton")
+	set_trait(TRAIT_PRODUCT_COLOUR,"#ffffff")
+	set_trait(TRAIT_PLANT_ICON,"bush2")
+	set_trait(TRAIT_WATER_CONSUMPTION, 6)
+
+/obj/item/seeds/cotton
+	seed_type = "cotton"
+
+/obj/item/reagent_containers/food/snacks/grown/attackby(obj/item/W, mob/user)
+	if(seed?.chems)
+		if(istype(W, /obj/item/loom))
+			if(!isnull(seed.chems[/datum/reagent/cottonfiber]))
+				user.visible_message(SPAN_NOTICE("\The [user] weaves the cotton into cloth \the [src]."))
```

Comment:
This grammatically doesn't make sense. If from a cotton plant, you'd have "[user] weaves the cotton into cloth cotton", but "[user] weaves the cotton into cloth potato" if xenoflora do magic.
Would 
```suggestion
				user.visible_message(SPAN_NOTICE("\The [user] weaves \the [src] into cotton cloth"))
```
be a better fit?